scripts/classification/FastTextGluon.py

```diff
@@ -0,0 +1,265 @@
+###############################################################################
+def train(args):	# Load and clean data
+    train_file = args.input #"/home/ubuntu/fastText/data/dbpedia.train"
+    test_file = args.validation #"/home/ubuntu/fastText/data/dbpedia.test"
+    ngram_range = args.ngrams
+    logging.info("Ngrams range for the training run : %s",ngram_range)
+    print("Loading Training data")
+    labels, data = read_input_data(train_file)
+    tokens_list = []
+    for x in data:
+        tokens_list.extend(x.split())
+        
+    cntr = Counter(tokens_list)
+    train_vocab = gluonnlp.Vocab(cntr)
+    print("Vocabulary size:",len(train_vocab))
+    print("Training data converting to sequences...")
+    train_sequences = [train_vocab.to_indices(x.split()) for x in data]
+    train_labels = labels
+    print("Reading test dataset")
+    test_labels, test_data = read_input_data(test_file)
+    test_sequences = [train_vocab.to_indices(x.split()) for x in test_data]
+
+    if ngram_range >= 2:
+        print('Adding {}-gram features'.format(ngram_range))
+        # Create set of unique n-gram from the training set.
+        ngram_set = set()
+        for input_list in train_sequences:
+            for i in range(2, ngram_range + 1):
+                    set_of_ngram = create_ngram_set(input_list, ngram_value=i)
+                    ngram_set.update(set_of_ngram)
+        start_index = len(cntr)
+        token_indices = {v: k + start_index for k, v in enumerate(ngram_set)}	    
+        train_sequences = add_ngram(train_sequences, token_indices, ngram_range)
+        test_sequences = add_ngram(test_sequences, token_indices, ngram_range)
+        print("Added n-gram features to train and test datasets!! ")
+    print("Encoding labels")
+    lb = LabelBinarizer()
+    lb.fit(train_labels)
+    encoded_y_train = lb.transform(train_labels)
+    encoded_y_test = lb.transform(test_labels)
+    y_train_final = encoded_y_train
+    y_test_final = encoded_y_test
+    #y_train_final = to_categorical(encoded_y_train)
+    #y_test_final = to_categorical(encoded_y_test)
+    n_labels = len(np.unique(train_labels))
+    print("Number of labels:",n_labels)
+    print("Initializing network")
+    ctx = mx.gpu(0)
+    embedding_dim = args.emsize	
+    num_classes = len(np.unique(train_labels))
+    print("Number of labels :",num_classes)
+    net = FastTextClassificationModel(len(train_vocab), 100, num_classes)
+    net.hybridize()
+    net.collect_params().initialize(mx.init.Xavier(), ctx = ctx )
+    print("Network initialized")
+
+    softmax_cross_entropy = gluon.loss.SoftmaxCrossEntropyLoss(sparse_label=False)    
+
+    num_epochs = args.epochs
+    batch_size = args.batch_size
+    print("Starting Training!")
+    learning_rate = args.lr
+    trainer1 = gluon.Trainer(net.embedding.collect_params(), 'adam', {'learning_rate': learning_rate})
+    trainer2 = gluon.Trainer(net.dense.collect_params(), 'adam', {'learning_rate': learning_rate})
```

Comment:
No need to use 2 trainers when using the same hyperparameters. You can just do:
`trainer = gluon.Trainer(net.collect_params(), 'adam', {'learning_rate': learning_rate})`